runelite-client/src/main/java/net/runelite/client/config/ConfigManager.java

```diff
@@ -175,24 +178,52 @@ public void load()
 		{
 			log.warn("Unable to update configuration on disk", ex);
 		}
+
+		if (configWatcherTask != null)
+		{
+			configWatcherTask.cancel(true);
+		}
+
+		configWatcherTask = executor.scheduleAtFixedRate(new ConfigWatcher(propertiesFile, () ->
+		{
+			final Properties properties = new Properties();
+			loadProperties(properties);
+
+			Map<String, String> copy = (Map) ImmutableMap.copyOf(properties);
+			copy.forEach((groupAndKey, value) ->
+			{
+				final String[] split = groupAndKey.split("\\.", 2);
+				if (split.length != 2)
+				{
+					log.debug("Properties key malformed!: {}", groupAndKey);
+					properties.remove(groupAndKey);
+					return;
+				}
+
+				final String groupName = split[0];
+				final String key = split[1];
+				final String oldValue = (String) this.properties.setProperty(groupName + "." + key, value);
```

Comment:
I guess our existing config changes are mostly on the AWT thread? Maybe not in this, but at some point we should lock around the config changes maybe or otherwise atomically apply them since they are accessed often by the client thread and assumed to not change.

Reply:
Hmm yeah a lot of the updating is from AWT thread (- like bank tags what is purely on client thread)